src/test/java/gov/nih/nci/evs/api/controller/SearchControllerTests.java

```diff
@@ -4015,6 +4021,115 @@ public void testSparqlPaging() throws Exception {
+    long total = 0;
+    final List<String> codes = new ArrayList<>();
+    while (true) {
+      log.info(
+          "Testing url - "
+              + url
+              + "?terminology=ncit&fromRecord="
+              + fromRecord
+              + "&pageSize="
+              + pageSize);
+      // Test a basic term search
+      result =
+          this.mvc
+              .perform(
+                  get(url)
+                      .param("terminology", "ncit")
+                      .param("fromRecord", "" + fromRecord)
+                      .param("pageSize", "" + pageSize))
+              .andExpect(status().isOk())
+              .andReturn();
+      content = result.getResponse().getContentAsString();
+      assertThat(content).isNotNull();
+      list = new ObjectMapper().readValue(content, ConceptResultList.class);
+      total = list.getTotal();
+      if (list.getConcepts() == null || list.getConcepts().isEmpty()) {
+        log.info("  done reading");
+        break;
+      }
+      codes.addAll(list.getConcepts().stream().map(c -> c.getCode()).collect(Collectors.toList()));
+      fromRecord += pageSize;
+    }
+
+    // Verify no duplicate codes
+    final Set<String> codeSet = new HashSet<>(codes);
+    assertThat(codeSet.size()).isEqualTo(codes.size());
+    assertThat(total).isEqualTo(codeSet.size());
+    assertThat((long) fromRecord).isGreaterThan(total);
+  }
+
+  /**
+   * Test searct all ncit with sort.
+   *
+   * @throws Exception the exception
+   */
+  @Test
+  public void testSearctAllNcitWithSort() throws Exception {
+    String url = null;
+    MvcResult result = null;
+    String content = null;
+    ConceptResultList list = null;
+
+    url = baseUrl;
+    int fromRecord = 0;
+    int pageSize = 500;
+    long total = 0;
+    final List<String> codes = new ArrayList<>();
+    while (true) {
+      log.info(
+          "Testing url - "
+              + url
+              + "?terminology=ncit&sort=code&fromRecord="
+              + fromRecord
+              + "&pageSize="
+              + pageSize);
+      // Test a basic term search
+      result =
+          this.mvc
+              .perform(
+                  get(url)
+                      .param("sort", "code")
+                      .param("terminology", "ncit")
+                      .param("fromRecord", "" + fromRecord)
+                      .param("pageSize", "" + pageSize))
+              .andExpect(status().isOk())
+              .andReturn();
+      content = result.getResponse().getContentAsString();
+      assertThat(content).isNotNull();
+      list = new ObjectMapper().readValue(content, ConceptResultList.class);
+      total = list.getTotal();
+      if (list.getConcepts() == null || list.getConcepts().isEmpty()) {
+        log.info("  done reading");
+        break;
+      }
+      codes.addAll(list.getConcepts().stream().map(c -> c.getCode()).collect(Collectors.toList()));
+      fromRecord += pageSize;
+    }
+
+    // Verify no duplicate codes
```

Comment:
Is this supposed to test that the sort parameter worked properly? Right now it seems to be doing the exact same checks as the other added test with the only difference being the sort parameter, which isn't checked.

Reply:
Good question. I'll put a comment to explain this. We don't care about checking sort order here, but a search without a sort parameter uses "rank" sort, so I'm checking that both "rank" sort and a chosen fielded sort both work properly (as I've seen situations with elasticsearch where they behave differently)